modin/core/io/text/csv_dispatcher.py

```diff
@@ -13,442 +13,12 @@
-            doesn't contain header rows)
-        pre_reading : int
-            The number of rows that should be read before data file
-            splitting for further reading (the number of rows for
-            the first partition).
-        skiprows_partitioning : int
-            The number of rows that should be skipped virtually (skipped during
-            data file partitioning).
-
-        Examples
-        --------
-        Let's consider case when `header`="infer" and `skiprows`=[3,4,5]. In
-        this specific case fastpath can be done since `skiprows` is uniformly
-        distributed array, so we can "squash" it to integer and set
-        `skiprows_partitioning`=3. But if no additional action will be done,
-        these three rows will be skipped right after header line, that corresponds
-        to `skiprows`=[1,2,3]. Now, to avoid this discrepancy, we need to assign
-        the first partition to read data between header line and the first
-        row for skipping by setting of `pre_reading` parameter, so setting
-        `pre_reading`=2. During data file partitioning, these lines will be assigned
-        for reading for the first partition, and then file position will be set at
-        the beginning of rows that should be skipped by `skiprows_partitioning`.
-        After skipping of these rows, the rest data will be divided between the
-        rest of partitions, see rows assignment below:
-
-        0 - header line (skip during partitioning)
-        1 - pre_reading (assign to read by the first partition)
-        2 - pre_reading (assign to read by the first partition)
-        3 - skiprows_partitioning (skip during partitioning)
-        4 - skiprows_partitioning (skip during partitioning)
-        5 - skiprows_partitioning (skip during partitioning)
-        6 - data to partition (divide between the rest of partitions)
-        7 - data to partition (divide between the rest of partitions)
-        """
-        pre_reading = 0
-        skiprows_partitioning = 0
-        skiprows_md = 0
-        if isinstance(skiprows, int):
-            skiprows_partitioning = skiprows
-        elif is_list_like(skiprows):
-            skiprows_md = np.sort(skiprows)
-            if np.all(np.diff(skiprows_md) == 1):
-                # `skiprows` is uniformly distributed array.
-                pre_reading = (
-                    skiprows_md[0] - header_size if skiprows_md[0] > header_size else 0
-                )
-                skiprows_partitioning = len(skiprows_md)
-                skiprows_md = 0
-            else:
-                skiprows_md = skiprows_md - header_size
-        elif callable(skiprows):
-
-            def skiprows_func(x):
-                return skiprows(x + header_size)
-
-            skiprows_md = skiprows_func
+    """Class handles utils for reading `.csv` files."""
 
-        return skiprows_md, pre_reading, skiprows_partitioning
+    read_callback = pandas.read_csv
```

Comment:
Why introduce another level of distinction? If necessary, `read_callback` can be used for pyarrow.

Globally, the idea is to describe mechanisms more generally so that different (engines+storage_formats) combinations can reuse them.